### DISCUSSION TITLE
Trouble setting up secure server

### DISCUSSION THREAD

**Discussion by seanrockster** [ravendb/ravendb#19681] [Q&A]
Hi

I've used RavenDb before however I'm attempting to setup a secure server (not done that before). So, i've followed the instructions as best I can but ultimately its not going as expected.

This is on an azure vm, RavenDb 6.2.1

I've downloaded Ravendb 6.2.1, ran the setup-as-service ps1 script, followed all the prompts, I've selected external and added the server ip address, 443 for the port, the certificate is created and downloaded. I've unzipped the certificate zip file (its in its own folder), run the admin certificate wizard, when prompted for personal or computer i've tried it both ways, after restarting the server it redirects to my url. Here is where it stops and just times out. If i try to to access the studio directly on the vm I get an authentication error however if I change the certificate i can see the studio but still shows an ssl error in the address bar. The url that was created during the setup is not accessible.

I've got a network security group attached to the vm and i have port 443 opened.

Wondering if anyone can suggest what I'm missing or what I can try to find the reason the generated *.community url doesn't work.

Thanks.

**Comment by gregolsky** [comment]
Can you provide your settings.json file please? Also does your VM have a public IP and if so can you provide it please?
Please make sure to redact all your secrets from the settings.

It might be that you set the .community DNS to something that is not accessible from your local IP address. Do you know what you entered there?

**Reply by seanrockster** [reply]
Hi

meanwhile I've got it working. I think the key was during selecting ip's and ports etc, i was using 127.0.0.1 and for external i was using the vm's public ip. I even tried the 0.0.0.0 option neither worked, however there was a 3rd ip which i'm assuming is an internal ip, but that worked and I can now issue certificates and access the sever externally. Thanks for your response.

**Reply by gregolsky** [reply]
Sean, that's great. Thanks for the update.

In most cases `ServerUrl` should be `https://0.0.0.0:443` and `PublicServerUrl` should be pointing to the VM's public IP and this is what the DNS should point to as well.
